ice/recipes/experiments_and_arms/prompts/passages_to_keep.py

```diff
@@ -132,7 +132,16 @@ async def get_best_paras(
         next_para = later_passages[0]
         later_passages = later_passages[1:]
         best_paras = best_paras | set((await get_best_paras([next_para])))
-    return [p for p in best_paras]
+
+    # Sort so that the returned order is deterministic for caching etc.
+    best_paras_sorted = sorted(best_paras)
```

Comment:
How about this instead of a separate reversal step?

```suggestion
    best_paras_sorted = sorted(best_paras, reverse=True)
```

Reply:
The separation is intentional. Sorting is important, reversing is a hacky workaround that should be removed. It also makes the comments nicer.

Reply:
The sorting also feels like a hacky workaround to me since it makes the order seem less arbitrary than it actually is. I'd mildly prefer a deterministic shuffle to make it clearer that we're okay with almost any arbitrary order as long as it's deterministic, and that lexicographic order isn't actually needed by this task, but I'm good with this as-is